File: iosApp/iosApp/MainViewModel.swift
//
//  MainViewModel.swift
//  iosApp
//
//  Created by Santiago Mattiauda on 7/4/24.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import home
import checkout
import data

class MainViewModel {
    
    let processCheckout = ProcessCheckout(repository: DataBookRepository())
    
    let getAll = GetAll(repository: DataBookRepository())
    
    
    func checkout() {
        let books = getAll.invoke()
        let currentBook = books.first
        
        //processCheckout.invoke(book: currentBook)
    }
}

File: androidApp/src/main/java/io/github/santimattius/kmp/android/MainViewModel.kt
package io.github.santimattius.kmp.android

import io.github.santimattius.kmp.data.Book
import io.github.santimattius.kmp.data.BookRepository
import io.github.santimattius.kmp.checkout.ProcessCheckout
import io.github.santimattius.kmp.home.GetAll

class MainViewModel {
    private val repository = BookRepository()
    private val processCheckout = ProcessCheckout(repository)
    private val getAll = GetAll(repository)

    fun checkout() {
        val books = getAll.invoke()
        val book: Book = books.first()

        processCheckout.invoke(book)
    }
}